spring-kafka/src/main/java/org/springframework/kafka/config/MethodKafkaListenerEndpoint.java

```diff
@@ -114,10 +115,26 @@ protected HandlerAdapter configureListenerAdapter(MessagingMessageListenerAdapte
 
 	/**
 	 * Create an empty {@link MessagingMessageListenerAdapter} instance.
+	 * @param messageConverter the converter (may be null).
 	 * @return the {@link MessagingMessageListenerAdapter} instance.
 	 */
-	protected MessagingMessageListenerAdapter<K, V> createMessageListenerInstance() {
-		return new MessagingMessageListenerAdapter<K, V>(this.method);
+	protected MessagingMessageListenerAdapter<K, V> createMessageListenerInstance(MessageConverter messageConverter) {
+		if (isBatchListener()) {
+			BatchMessagingMessageListenerAdapter<K, V> messageListener = new BatchMessagingMessageListenerAdapter<K, V>(
+					this.method);
```

Comment:
That sounds reasonable and like a gold compromise.
Let's leave as is! Even without annotation changes.